precompiles/act/tx.go

```diff
@@ -0,0 +1,288 @@
+package act
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	cmn "github.com/evmos/evmos/v20/precompiles/common"
+	"github.com/evmos/evmos/v20/x/evm/core/vm"
+	precommon "github.com/warden-protocol/wardenprotocol/precompiles/common"
+	actmodulekeeper "github.com/warden-protocol/wardenprotocol/warden/x/act/keeper"
+	types "github.com/warden-protocol/wardenprotocol/warden/x/act/types/v1beta1"
+)
+
+const (
+	CheckActionMethod    = "checkAction"
+	NewTemplateMethod    = "newTemplate"
+	RevokeActionMethod   = "revokeAction"
+	UpdateTemplateMethod = "updateTemplate"
+	VoteForActionMethod  = "voteForAction"
+)
+
+func (p Precompile) CheckActionMethod(
+	ctx sdk.Context,
+	origin common.Address,
+	stateDB vm.StateDB,
+	method *abi.Method,
+	args []interface{},
+) ([]byte, error) {
+	msgServer := actmodulekeeper.NewMsgServerImpl(p.actmodulekeeper)
+
+	message, err := newMsgCheckAction(args, origin)
+
+	if err != nil {
+		return nil, err
+	}
+
+	p.Logger(ctx).Debug(
+		"tx called",
+		"method", method.Name,
+		"args", fmt.Sprintf(
+			"{ action_id: %d, creator: %s }",
+			message.ActionId,
+			message.Creator,
+		),
+	)
```

Comment:
it should basically be the same as
```suggestion
	p.Logger(ctx).Debug(
		"tx called",
		"method", method.Name,
		"args", message,
		),
	)
```
the small improvement is that we don't run fmt.Sprintf when debug logs are disabled